### DISCUSSION TITLE
Have any ideas for GeminiPilot? Drop a comment here!

### DISCUSSION THREAD

**Discussion by HenryGetz** [HenryGetz/GeminiPilot#1] [Ideas]
This is the place for all of your great ideas about this userscript. Want another shortcut, or something else? Let me know here!

**Comment by joshmoore1** [comment]
I personally like being able to jump between _prompts_, and I made a simple version that uses PageUp and PageDown to jump to the nearest one. It works between two prompts as well, just jumps up or down to the nearest one. It also expands them, as usually I'm trying to read that. 

Maybe some people use PageUp and PageDown normally within Gemini, and would want Ctrl or Shift+PageUp for example, or there could be better shortcut buttons, up to you. Could also make the "Expand text" feature optional with Ctrl or Shift. I left it as the default as I'm almost always jumping between prompts trying to read them and remember what each one was, so its worth scrolling down past the expanded prompt a little bit more when I find the one I was looking for. Maybe for super long prompts a way to collapse the current text could be helpful? Just figured I'd suggest these, as I thought of them previously. Also figured I'd suggest this idea in general as I was looking for this feature and it brought me to your script!

I attached the most recent version of mine below. It was a simple test, could probably be improved using .closest or something, but it works reliably and seems to be robust for detecting the html elements of the prompts as of 9/3/25. If you want me to reply to this comment with the raw code instead I can, just didn't want to clutter the comment.

[Gemini Prompt Navigator-1.0.user.js](https://github.com/user-attachments/files/22122245/Gemini.Prompt.Navigator-1.0.user.js)

**Comment by joshmoore1** [comment]
One other idea that could be helpful. I really like how it expands the width of the chat window depending on the size of your browser window. I don't know why that isn't Gemini's default, honestly. Could you also make it expand the width of the prompts? Currently, they stay as thin as they were before, which is pretty thin in landscape on a 1920x1080 monitor. (Maybe only 25% of the screen width?)

**Comment by clackwork** [comment]
Great job on the extension! i created the following issue and perhaps i should have just dropped my feedback in the discussion. please feel free to close it.
- https://github.com/HenryGetz/GeminiPilot/issues/17

Issue title: "feature: Have tampermonkey script automatically security reviewed by external 3rd party".

**Comment by clackwork** [comment]
Is this tested at all on Firefox? A lot of keyboard shortcuts and other stuff doesn't work. I still get value from the "enable microphone" shortcut, so that makes me happy. For instance, shift-escape is shadowed by the pre-existing "open process manager" global shortcut.
